adam-core/src/main/scala/org/bdgenomics/adam/converters/FastqRecordConverter.scala

```diff
@@ -143,10 +145,25 @@ class FastqRecordConverter extends Serializable with Logging {
     // get fields for first read in pair
     val readName = trimTrailingReadNumber(lines(0).drop(1))
     val readSequence = lines(1)
-    val readQualities = lines(3)
+
+    if (stringency == ValidationStringency.STRICT && lines(3) == "*" && readSequence.length > 1)
+      throw new Exception(s"Fastq quality must be defined")
+    else if (stringency == ValidationStringency.STRICT && lines(3).length != readSequence.length)
+      throw new Exception(s"Fastq sequence and quality strings must have the same length")
+
+    val readQualities =
```

Comment:
here's how HTSJDK handles it, I propose doing the same:
https://github.com/samtools/htsjdk/blob/master/src/java/htsjdk/samtools/ValidationStringency.java



Sent with MailTrack
<https://mailtrack.io/install?source=signature&lang=en&referral=allenday@allenday.com&idSignature=22>

On Tue, Oct 20, 2015 at 10:08 PM, Michael L Heuer <notifications@github.com>
wrote:

> In
> adam-core/src/main/scala/org/bdgenomics/adam/converters/FastqRecordConverter.scala
> <https://github.com/bigdatagenomics/adam/pull/856#discussion_r42508048>:
>
> > @@ -143,10 +145,25 @@ class FastqRecordConverter extends Serializable with Logging {
> >      // get fields for first read in pair
> >      val readName = trimTrailingReadNumber(lines(0).drop(1))
> >      val readSequence = lines(1)
> > -    val readQualities = lines(3)
> > +
> > +    if (stringency == ValidationStringency.STRICT && lines(3) == "*" && readSequence.length > 1)
> > +      throw new Exception(s"Fastq quality must be defined")
> > +    else if (stringency == ValidationStringency.STRICT && lines(3).length != readSequence.length)
> > +      throw new Exception(s"Fastq sequence and quality strings must have the same length")
> > +
> > +    val readQualities =
>
> I think rather than fixing invalid FASTQ records when reading them in with
> non-strict validation stringency, I would prefer they get logged as invalid
> and dropped. Then for strict validation stringency, encountering an invalid
> FASTQ record would throw an exception that causes the read operation to
> stop.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/bigdatagenomics/adam/pull/856/files#r42508048>.
>